### PR TITLE
Vote bar

### LINKED ISSUE #506: Better UX for progress bar in voting phase
The progress bar on the idea detail page in the voting phase currently shows red (no votes), green (yes votes) and yellow (implicit abstains = number of students in idea space). This seems to be confusing not just to me but also to Marina‘s sister. I would suggest two changes: 

1. Do not show the implicit abstains, i.e. the yellow part of the bar
1. If no votes were cast the bar is just and shows the “0 votes"

(v0.11.3)

---
Original text from the client: 

Unter besagter Leiste werden Stimmen dafür und dagegen richtig angezeigt, unter dem gelben Teil der Leiste steht allerdings völlig unabhängig von der Anzahl der gegebenen Stimmen immer die Nummer 20.

### PR DESCRIPTION
**Pull request by fisx**
Fixes #506, #522.

### CODE REVIEW

**Comment by fisx** [comment]
@localgrr 

```haskell
Frame {_frameUser = User {_userMeta = MetaInfo {_metaKey = AUID 22, _metaCreatedBy = AUID 6, _metaCreatedByLogin = UserLogin {_unUserLogin = "quaerat"}, _metaCreatedByAvatar = Nothing, _metaCreatedAt = 2016-05-18_13:51:22_391905000000, _metaChangedBy = AUID 6, _metaChangedAt = 2016-05-18_13:51:22_391905000000}, _userLogin = UserLogin {_unUserLogin = "iuscoa"}, _userFirstName = UserFirstName {_unUserFirstName = "iusto"}, _userLastName = UserLastName {_unUserLastName = "consequat"}, _userRole = Student {_roleSchoolClass = SchoolClass {_classSchoolYear = 2016, _className = "12b"}}, _userProfile = UserProfile {_profileAvatar = Just "/static/demo/avatars/test_user.png", _profileDesc = Markdown {unMarkdown = "loves commodo long consequatur resultant soluta pleasure Et pleasure natus small nulla welcomed pain sit last totam big No numquam sequi was Quis But voluptates\n\natque accepted one blinded nostrud facilis illo natus incididunt adipiscing adipisci perfectly repudiated magni again consequatur cannot\n\nwe velit our labore pain exercitationem maxime praising itself you provident do repudiated must possimus he painful corrupti commodi\n\n"}}, _userSettings = UserSettings {_userSettingsPassword = UserPassInitial {_userPassInitial = "Xc\b=\CANo\248"}, _userSettingsEmail = Nothing}}, _frameBody = ViewIdea (RenderContext {_renderContextUser = User {_userMeta = MetaInfo {_metaKey = AUID 22, _metaCreatedBy = AUID 6, _metaCreatedByLogin = UserLogin {_unUserLogin = "quaerat"}, _metaCreatedByAvatar = Nothing, _metaCreatedAt = 2016-05-18_13:51:22_391905000000, _metaChangedBy = AUID 6, _metaChangedAt = 2016-05-18_13:51:22_391905000000}, _userLogin = UserLogin {_unUserLogin = "iuscoa"}, _userFirstName = UserFirstName {_unUserFirstName = "iusto"}, _userLastName = UserLastName {_unUserLastName = "consequat"}, _userRole = Student {_roleSchoolClass = SchoolClass {_classSchoolYear = 2016, _className = "12b"}}, _userProfile = UserProfile {_profileAvatar = Just "/static/demo/avatars/test_user.png", _profileDesc = Markdown {unMarkdown = "loves commodo long consequatur resultant soluta pleasure Et pleasure natus small nulla welcomed pain sit last totam big No numquam sequi was Quis But voluptates\n\natque accepted one blinded nostrud facilis illo natus incididunt adipiscing adipisci perfectly repudiated magni again consequatur cannot\n\nwe velit our labore pain exercitationem maxime praising itself you provident do repudiated must possimus he painful corrupti commodi\n\n"}}, _userSettings = UserSettings {_userSettingsPassword = UserPassInitial {_userPassInitial = "Xc\b=\CANo\248"}, _userSettingsEmail = Nothing}}}) (IdeaStats {_listInfoForIdeaIt = Idea {_ideaMeta = MetaInfo {_metaKey = AUID 4, _metaCreatedBy = AUID 2, _metaCreatedByLogin = UserLogin {_unUserLogin = "godmin"}, _metaCreatedByAvatar = Nothing, _metaCreatedAt = 2016-05-18_13:51:22_371612000000, _metaChangedBy = AUID 2, _metaChangedAt = 2016-05-18_13:51:22_371612000000}, _ideaTitle = "topic-idea-title", _ideaDesc = Markdown {unMarkdown = "topic-idea-desc"}, _ideaCategory = Just CatRules, _ideaLocation = IdeaLocationTopic {_ideaLocationSpace = SchoolSpace, _ideaLocationTopicId = AUID 5}, _ideaComments = fromList [], _ideaLikes = fromList [], _ideaVotes = fromList [(AUID 22,IdeaVote {_ideaVoteMeta = MetaInfo {_metaKey = IdeaVoteLikeKey {_ivIdea = AUID 4, _ivUser = AUID 22}, _metaCreatedBy = AUID 22, _metaCreatedByLogin = UserLogin {_unUserLogin = "iuscoa"}, _metaCreatedByAvatar = Just "/static/demo/avatars/test_user.png", _metaCreatedAt = 2016-05-19_09:12:27_949016000000, _metaChangedBy = AUID 22, _metaChangedAt = 2016-05-19_09:12:27_949016000000}, _ideaVoteValue = Yes})], _ideaJuryResult = Just (IdeaJuryResult {_ideaJuryResultMeta = MetaInfo {_metaKey = AUID 2968, _metaCreatedBy = AUID 1, _metaCreatedByLogin = UserLogin {_unUserLogin = "admin"}, _metaCreatedByAvatar = Nothing, _metaCreatedAt = 2016-05-18_13:51:38_526611000000, _metaChangedBy = AUID 1, _metaChangedAt = 2016-05-18_13:51:38_526611000000}, _ideaJuryResultValue = Feasible {_ideaResultFeasibleReason = Nothing}}), _ideaVoteResult = Nothing}, _listInfoForIdeaPhase = PhaseVoting {_phaseStatus = ActivePhase {_phaseEnd = 2016-06-08_13:51:38_530145000000}}, _listInfoForIdeaQuorum = 39, _listInfoForIdeaNoOfVoters = 130}), _frameMessages = []}
```

File: src/Frontend/Fragment/VotesBar.hs [hunk 1]
```diff
@@ -41,49 +49,54 @@ instance ToHtml IdeaVoteLikeBars where
...
               where
-                yesVotes    = numVotes idea Yes ^. showed . html
-                noVotes     = numVotes idea No  ^. showed . html
-                yesPercent  = max (percentVotes idea voters Yes) 5 ^. showed
-                noPercent   = max (percentVotes idea voters No)  5 ^. showed
+                cnt :: IdeaVoteValue -> Html ()
+                cnt v = numVotes idea v ^. showed . html
+
+                prcnt :: IdeaVoteValue -> ST
+                prcnt v = max (percentVotes idea oneHundret v) 5 ^. showed . csi
```
Comment (andorp) [inline src/Frontend/Fragment/VotesBar.hs]: Please turn magic number `5` to a constant of this module.

File: src/Frontend/Fragment/VotesBar.hs [hunk 2]
```diff
@@ -18,6 +18,14 @@ import           Persistent.Idiom
 data IdeaVoteLikeBars = IdeaVoteLikeBars RenderContext [IdeaCapability] IdeaStats
   deriving (Eq, Show, Read)
 
+-- | The issue has been debated for some time now whether we should show three segments (yes, no,
+-- not voted) or just two (yes, no), we introduced a switch.
+data ShowNotVoted = ShowNotVoted | DoNotShowNotVoted
```
Comment (andorp) [inline src/Frontend/Fragment/VotesBar.hs]: I like mini DSLs :+1:

**Comment by andorp** [comment]
LGTM

**Pull request merged**